Emitting declarations for commonjs exported classes suggests classes are private, TS9006

<!-- 🚨 STOP 🚨 STOP 🚨 STOP 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!--
Please try to reproduce the issue with the latest published version. It may have already been fixed.
For npm: `typescript@next`
This is also the 'Nightly' version in the playground: http://www.typescriptlang.org/play/?ts=Nightly
-->
**TypeScript Version:**  4.0.3 (also happens in `typescript@next`)

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
private name,Declaration emit,block declaration emit,TS9006

a very simple code sample - 
https://github.com/euclideansphere/private-declaration-issue

```js
// utils/errors.js
class FancyError extends Error {
	constructor(status) {
		super(`error with status ${status}`);
	}
}

module.exports = {
	FancyError
}

// utils/index.js
// issue arises here on compilation
const errors = require("./errors");

module.exports = {
	errors
}
```

**Expected behavior:**

```sh
(master)⚡ % tsc # should run clean, exit with code 0
```

**Actual behavior:**

```sh
(master)⚡ % tsc                                                                                                                                                       ~/projects/private-declaration-issue
utils/index.js:1:1 - error TS9006: Declaration emit for this file requires using private name 'FancyError' from module '"/Users/austin/projects/private-declaration-issue/utils/errors"'. An explicit type annotation may unblock declaration emit.

1 const errors = require("./errors");
  ~~~~~


Found 1 error.
```

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior -->
https://github.com/euclideansphere/private-declaration-issue

**Related Issues:** <!-- Did you find other bugs that looked similar? -->
https://github.com/Microsoft/TypeScript/issues/9865 - seems to be an issue with "A inherits B but only A is exported", but in this case B is a global and adding it as an export doesn't resolve the issue.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -6800,7 +6800,9 @@ namespace ts {
                             }
                             // else fall through and treat commonjs require just like import=
                         case SyntaxKind.ImportEqualsDeclaration:
-                             if (target.escapedName === InternalSymbolName.ExportEquals) {
+                            // This _specifically_ only exists to handle json declarations - where we make aliases, but since
+                            // we emit no declarations for the json document, must not refer to it in the declarations
+                            if (target.escapedName === InternalSymbolName.ExportEquals && some(target.declarations, isJsonSourceFile)) {
                                 serializeMaybeAliasAssignment(symbol);
                                 break;
                             }

--- a/tests/baselines/reference/jsDeclarationsExportedClassAliases.js
+++ b/tests/baselines/reference/jsDeclarationsExportedClassAliases.js
@@ -1,0 +1,60 @@
+//// [tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportedClassAliases.ts] ////
+
+//// [errors.js]
+class FancyError extends Error {
+    constructor(status) {
+        super(`error with status ${status}`);
+    }
+}
+
+module.exports = {
+    FancyError
+};
+
+//// [index.js]
+// issue arises here on compilation
+const errors = require("./errors");
+
+module.exports = {
+    errors
+};
+
+//// [errors.js]
+var __extends = (this && this.__extends) || (function () {
+    var extendStatics = function (d, b) {
+        extendStatics = Object.setPrototypeOf ||
+            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
+            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
+        return extendStatics(d, b);
+    };
+    return function (d, b) {
+        extendStatics(d, b);
+        function __() { this.constructor = d; }
+        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
+    };
+})();
+var FancyError = /** @class */ (function (_super) {
+    __extends(FancyError, _super);
+    function FancyError(status) {
+        return _super.call(this, "error with status " + status) || this;
+    }
+    return FancyError;
+}(Error));
+module.exports = {
+    FancyError: FancyError
+};
+//// [index.js]
+// issue arises here on compilation
+var errors = require("./errors");
+module.exports = {
+    errors: errors
+};
+
+
+//// [errors.d.ts]
+export class FancyError extends Error {
+    constructor(status: any);
+}
+//// [index.d.ts]
+import errors = require("./errors");
+export { errors };

--- a/tests/baselines/reference/jsDeclarationsExportedClassAliases.symbols
+++ b/tests/baselines/reference/jsDeclarationsExportedClassAliases.symbols
@@ -1,0 +1,40 @@
+=== tests/cases/conformance/jsdoc/declarations/utils/index.js ===
+// issue arises here on compilation
+const errors = require("./errors");
+>errors : Symbol(errors, Decl(index.js, 1, 5))
+>require : Symbol(require)
+>"./errors" : Symbol("tests/cases/conformance/jsdoc/declarations/utils/errors", Decl(errors.js, 0, 0))
+
+module.exports = {
+>module.exports : Symbol("tests/cases/conformance/jsdoc/declarations/utils/index", Decl(index.js, 0, 0))
+>module : Symbol(export=, Decl(index.js, 1, 35))
+>exports : Symbol(export=, Decl(index.js, 1, 35))
+
+    errors
+>errors : Symbol(errors, Decl(index.js, 3, 18))
+
+};
+=== tests/cases/conformance/jsdoc/declarations/utils/errors.js ===
+class FancyError extends Error {
+>FancyError : Symbol(FancyError, Decl(errors.js, 0, 0))
+>Error : Symbol(Error, Decl(lib.es5.d.ts, --, --), Decl(lib.es5.d.ts, --, --))
+
+    constructor(status) {
+>status : Symbol(status, Decl(errors.js, 1, 16))
+
+        super(`error with status ${status}`);
+>super : Symbol(ErrorConstructor, Decl(lib.es5.d.ts, --, --))
+>status : Symbol(status, Decl(errors.js, 1, 16))
+    }
+}
+
+module.exports = {
+>module.exports : Symbol("tests/cases/conformance/jsdoc/declarations/utils/errors", Decl(errors.js, 0, 0))
+>module : Symbol(export=, Decl(errors.js, 4, 1))
+>exports : Symbol(export=, Decl(errors.js, 4, 1))
+
+    FancyError
+>FancyError : Symbol(FancyError, Decl(errors.js, 6, 18))
+
+};
+

--- a/tests/baselines/reference/jsDeclarationsExportedClassAliases.types
+++ b/tests/baselines/reference/jsDeclarationsExportedClassAliases.types
@@ -1,0 +1,47 @@
+=== tests/cases/conformance/jsdoc/declarations/utils/index.js ===
+// issue arises here on compilation
+const errors = require("./errors");
+>errors : { FancyError: typeof FancyError; }
+>require("./errors") : { FancyError: typeof FancyError; }
+>require : any
+>"./errors" : "./errors"
+
+module.exports = {
+>module.exports = {    errors} : { errors: { FancyError: typeof FancyError; }; }
+>module.exports : { errors: { FancyError: typeof FancyError; }; }
+>module : { "\"tests/cases/conformance/jsdoc/declarations/utils/index\"": { errors: { FancyError: typeof FancyError; }; }; }
+>exports : { errors: { FancyError: typeof FancyError; }; }
+>{    errors} : { errors: { FancyError: typeof FancyError; }; }
+
+    errors
+>errors : { FancyError: typeof FancyError; }
+
+};
+=== tests/cases/conformance/jsdoc/declarations/utils/errors.js ===
+class FancyError extends Error {
+>FancyError : FancyError
+>Error : Error
+
+    constructor(status) {
+>status : any
+
+        super(`error with status ${status}`);
+>super(`error with status ${status}`) : void
+>super : ErrorConstructor
+>`error with status ${status}` : string
+>status : any
+    }
+}
+
+module.exports = {
+>module.exports = {    FancyError} : { FancyError: typeof FancyError; }
+>module.exports : { FancyError: typeof FancyError; }
+>module : { "\"tests/cases/conformance/jsdoc/declarations/utils/errors\"": { FancyError: typeof FancyError; }; }
+>exports : { FancyError: typeof FancyError; }
+>{    FancyError} : { FancyError: typeof FancyError; }
+
+    FancyError
+>FancyError : typeof FancyError
+
+};
+

--- a/tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportedClassAliases.ts
+++ b/tests/cases/conformance/jsdoc/declarations/jsDeclarationsExportedClassAliases.ts
@@ -1,0 +1,22 @@
+// @allowJs: true
+// @checkJs: true
+// @outDir: ./out
+// @declaration: true
+// @filename: utils/errors.js
+class FancyError extends Error {
+    constructor(status) {
+        super(`error with status ${status}`);
+    }
+}
+
+module.exports = {
+    FancyError
+};
+
+// @filename: utils/index.js
+// issue arises here on compilation
+const errors = require("./errors");
+
+module.exports = {
+    errors
+};

